numbers concept

colinleach opened the pull request:
This is an unfinished draft, NOT ready for formal review.

Known limitations:

- `introduction.md` and `links.json` are both empty
- References within `about.md` are very Wikipedia-heavy. I need to find more Julia-specific links.
- We need to decide what exercise to pair this with, before writing `introduction.md`

I'm pushing this to `concepts.wip` to get it up on GH for discussion. Also as an indication of the way I'm thinking (rightly or wrongly) about the syllabus. Also as I knind of apology for pushing back quite aggressively against adding more content to `basics` - I wanted to discuss things properly in this separate concept.

colinleach commented:
P.S. It's long!

What do you expect, numbers are what Julia is _for_.

I'm assuming some level of mathematical knowledge, and will defend that. This language is unlikely to attract front-end developers from the JS/HTML/CSS world, it's designed for a different purpose and biased towards a different user group.

However, I think my text should be a _lot_ more approachable than much of the official documentation!

colinleach commented:
@depial, great to see you back! I guessed you might be within range of Helene's effects (if so you have my sympathy - Sandy in 2012 was more than enough for me, in my PA-resident days), so I've tried to avoid distracting you further.

I just read the "mathy" criticism. Two points I would make in pushing back:
- This is an `about.md` where students can skip through the bits that don't suit them. It should be less controversial than an exercise that forces students to engage with the mathy stuff. Though the management are going to _love_ the Linear Algebra concept!
- This is Julia, _of course_ it's going to be mathy!!!! Just like R is statistics-heavy, and JS is webby.

I know that @BethanyG has been keen for a long time to nudge the scientific languages in this direction. That's hard in Python, because it requires adding big 3rd-party packages to the test runner (NumPy for starters). In Julia it's the core language, and central to the design from its earliest days.

Our (untested) theory is that JS/TS/Elm users are unlikely to engage much with tracks like Julia and R, but the students who do are probably quite interested in the mathy stuff if we introduce it gently. Not all Exercism users are the same _(says the rather untypical 67-year-old user with a PhD in chemistry)_.

OTOH, Jeremy has to pay the server bills each month, so he gets the final word on what they run.

depial commented:
> I guessed you might be within range of Helene's effects (if so you have my sympathy - Sandy in 2012 was more than enough for me, in my PA-resident days)

Thanks for your concern! Seeing/Hearing what's happened all around us, there's no way I can complain. We got really lucky here for some reason.

> This is an about.md where students can skip through the bits that don't suit them.

That seems more easily accepted indeed. (I still need to get a handle on the structure of everything which I hope to do through reading your PRs.)

> Our (untested) theory is that JS/TS/Elm users are unlikely to engage much with tracks like Julia and R, but the students who do are probably quite interested in the mathy stuff if we introduce it gently. Not all Exercism users are the same (says the rather untypical 67-year-old user with a PhD in chemistry).

I'm fully on board with this. Not only in the idealistic sense that everyone should have access to relevant education (my favorite sense by all means), but it also makes business sense to have diversity in your platform since is builds robustness against the rather fickle nature of "the market". I feel like a lot of sites/businesses go too far down the standardization route, end up disenfranchising a good portion of their most loyal fanbase and then find themselves scrambling to cope when trends changed. I think Exercism is interesting because it seems like both of these senses are equally relevant, where usually its only the latter.

One last thing... I'm not sure if you saw Jeremy's change to the introduction, but it was the most encouraging bit of the review process:

> **Note: This is quite a _mathsy_ exercise. If you don't enjoy solving maths-based exercises, maybe skip this one. But if you do, you'll enjoy this.**

So *mathiness* doesn't seem to be an automatic down vote.

colinleach commented:
> I hope to start to be able to look through and comment on some things you've been doing (it's quite a bit!)

I started out trying to be quite systematic, getting each concept/exercise pair merged before moving on to the next one. That got bogged down in lots of slow and detailed word-smithing. Fine - this made the documents much better. However, it stopped me getting a feel for the overall shape of the syllabus.

For the last couple of weeks, I've been posting quite rough early drafts of concepts, to see how things evolve. They will need significant improvement before eventually being merged, but I'm finding the exercise very educational.

I recently came across a quote that Julia is quite a "big" language with lots built in (I can't find the text right now). My current guess it that a syllabus will need _at least_ 40 concepts, maybe significantly more. Settle in for a long slog...

colinleach commented:
Continuing the easy ones, I'll try to get `pairs-and-dicts` posted tomorrow and `sets` soon after. Both have precedent in other tracks, so we can steal the exercises.

colinleach commented:
I think this is about ready for review, though push back if you disagree. I moved it from `concepts.wip/` to `concepts/` and fixed various small things (references, `config.sys` entry).

colinleach commented:
Following [advice from @glennj](http://forum.exercism.org/t/is-there-a-way-to-deprecate-old-julia-concepts/15999/2), I've changed the `concept.json` entry to reuse an old UUID from a concept (comments) that we want to get rid of. To placate the linter, I also had to move `concepts/comments/` to `concepts.wip/comments/`. CI now seems happy with it, but the website plays to its own rules.

Now I'll wait nervously to see what I broke this time...

depial reviewed (commented):
Just one minor nit that popped out all of a sudden. Not sure if it's anything to be honest.

colinleach commented on concepts/numbers/introduction.md:
```diff
@@ -0,0 +1,203 @@
+
+## Floating-point
+
+It will be no surprise that floating-point numbers optionally have a decimal point, and a fractional part after the point.
```
Let's remove the word "optionally". I just did a few experiments, and have not yet managed to suppress the display of the point. It may be just my ignorance, though.

```julia-repl
julia> round(7.0, digits=0)
7.0
```

colinleach commented:
Thank you! I just checked the website, and the Comments concept has disappeared from the Learning Track. Numbers should appear once Currency Exchange is merged.

I'm currently trying to sort out Basics and the new version of Lasagna, so the Learning tree has a root node. Very confusing, they have too much history to grapple with!
